CodeJam.Main/Assertions/ArgumentAssertionExtensions.cs

```diff
@@ -56,44 +55,44 @@ public static class ArgumentAssertionExtensions
 		}
 
 		/// <summary>
-		/// Ensures that supplied enumerable is not empty.
+		/// Ensures that supplied enumerable is not null nor empty.
 		/// </summary>
 		/// <typeparam name="T">Type of item.</typeparam>
 		/// <param name="arg">The argument.</param>
 		/// <returns><paramref name="arg"/></returns>
 		[DebuggerHidden, MethodImpl(AggressiveInlining)]
 		[AssertionMethod]
```

Comment:
Shouldnt add contractannotation here ?

Reply:
[Contract annotations](https://www.jetbrains.com/help/resharper/Contract_Annotations.html#syntax) do not support reference to the item's value so there's no way to express "arg.Argument should be not null" contract

Reply:
And for arg itself?
arg:null => halt

Reply:
It will has no effect as ArgumentAssertion<T> is a struct.